### PR TITLE
Add data compression to PositionForwarderWialon

### PR DESCRIPTION
**Pull request by TheArrex**
If data is bigger then MTU, UDP won't allow to send this. I try to change UDP to TCP, chunk data, but compression do it's best.
Compression is described at the end of this docs: https://extapi.wialon.com/hw/cfg/Wialon%20IPS_en_v_2_0.pdf
Tested on Wialon and build looks fine.

### CODE REVIEW

**Comment by tananaev** [comment]
1. Can you rebase on master properly. I see duplicated changes that already exist on the master branch.
2. It doesn't look like Wialon 1.x supports zip.
3. We also don't support zip in our server for Wialon.

File: src/main/java/org/traccar/forward/PositionForwarderWialon.java [hunk 1]
```diff
@@ -111,28 +116,56 @@ public void forward(PositionData positionData, ResultHandler resultHandler) {
...
+        container.write(0xFF);
+
+        int dataLength = data.length;
+        ByteBuffer lengthBuffer = ByteBuffer.allocate(2).order(ByteOrder.LITTLE_ENDIAN);
+        lengthBuffer.putShort((short) dataLength);
+        container.write(lengthBuffer.array(), 0, 2);
+
+        container.write(data, 0, dataLength);
+
+        return container.toByteArray();
```
Comment (tananaev) [inline src/main/java/org/traccar/forward/PositionForwarderWialon.java:168]: Using Netty buffers can probably make this much cleaner.

**Comment by TheArrex** [comment]
> 2. It doesn't look like Wialon 1.x supports zip.
> 3. We also don't support zip in our server for Wialon.

Should i just wrap this feature with `version.startsWith("2")` condition?

And maybe we need to change protocol version from hardcode at MainModule to config option?

**Comment by tananaev** [comment]
I think it would be better to just have a constructor parameter to enable/disable ZIP.

**Comment by TheArrex** [comment]
> I think it would be better to just have a constructor parameter to enable/disable ZIP.

Disabled by default?

**Comment by tananaev** [comment]
Yes, I prefer disabled by default because then the format will be compatible with Traccar.

File: src/main/java/org/traccar/forward/PositionForwarderWialon.java [hunk 2]
```diff
@@ -111,6 +129,33 @@ public void forward(PositionData positionData, ResultHandler resultHandler) {
         }
     }
 
+    public static byte[] compressData(byte[] data) {
```
Comment (tananaev) [inline src/main/java/org/traccar/forward/PositionForwarderWialon.java:132]: I would probably just combine the two methods to avoid two buffers.

**Pull request merged**